core/src/files.rs

```diff
@@ -117,7 +117,7 @@ impl Files {
         // This implementation would be a little nicer if `Vector` supported mutable access.
         // unwrap: we're allowed to panic if file_id is invalid
         let mut old = self.get(file_id).unwrap().clone();
-        old.source = source.into();
+        old = File::new(old.name, source);
```

Comment:
This fixes a recently-introduced but where the `line_starts` cache wasn't updated. Unlikely to trigger in `nickel` itself, but it caused some problems for nls.